src/main/kotlin/apply/application/MailHistoryService.kt

```diff
@@ -0,0 +1,45 @@
+package apply.application
+
+import apply.application.mail.MailData
+import apply.domain.mail.MailHistory
+import apply.domain.mail.MailHistoryRepository
+import apply.domain.mail.getById
+import org.springframework.stereotype.Service
+import javax.transaction.Transactional
+
+@Transactional
+@Service
+class MailHistoryService(
+    private val mailHistoryRepository: MailHistoryRepository
+) {
+    fun save(mailData: MailData) {
+        mailHistoryRepository.save(
+            MailHistory(
+                mailData.subject,
+                mailData.body,
+                mailData.sender,
+                mailData.recipientsToString(),
+                mailData.sentTime
+            )
+        )
+    }
+
+    fun findAll(): List<MailData> {
+        val emailHistory = mailHistoryRepository.findAll()
+        return emailHistory.map {
+            MailData(
+                it.subject,
+                it.body,
+                it.sender,
+                it.recipients,
+                it.sentTime,
+                it.id
+            )
+        }
```

Comment:
재사용하지 않는 지역 변수의 수는 최소화하면 어떨까요? `MailData`에 부 생성자를 추가할 수도 있어요.

```suggestion
        return mailHistoryRepository.findAll()
            .map { MailData(it.subject, it.body, it.sender, it.recipients, it.sentTime, it.id) }
```

Reply:
이미 부 생성자를 만들어 놨었는데 활용하지 못하고 있었네요! 수정하였습니다~ㅎㅎㅎ